src/artifactorypurge/artifactorypurge.py

```diff
@@ -1,64 +1,81 @@
+from .utils.logging import getLogger
 from .utils.performance import get_performance_report
 
 LOG = getLogger(__name__)
 
+
 @click.command()
-@click.option('--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
-@click.option('--reponame', default=None, help='Operate on single repo')
-@click.option('--project', default=None, help='Operate on single project')
+@click.option(
+    '--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
+@click.option('--default/--no-default', default=True, is_flag=True, help='If false, does not apply default policy')
+@click.option('--policies-path', required=False, help='Path to extra policies directory')
 #@click.argument('url')
-def purge(dryrun, reponame, project): #, url):
-    exceptions = {}
-
+def purge(dryrun, policies_path, default):  #, url):
     credentials = load_credentials()
-    artifactory = Artifactory(credentials['artifactory_url'], 
-                              credentials['artifactory_username'],
+    artifactory = Artifactory(credentials['artifactory_url'], credentials['artifactory_username'],
                               credentials['artifactory_password'])
 
-    before = artifactory.list(reponame)
+    plugin_source = setup_pluginbase(extra_policies_path=policies_path)
+    before = artifactory.list(None)
     for repo, info in before.items():
-        fp = 0
+        policy_name = repo.replace("-", "_")
         try:
-            modulename = repo.replace("-", "_")
-            fp, pathname, description = imp.find_module(modulename, ["repositories",])
-            module =  imp.load_module(modulename, fp, pathname, description)
-            artifacts = module.purgelist(artifactory, repo, project)
-            count = artifactory.purge(repo, dryrun, artifacts)
-            LOG.info("processed {}, purged {}".format(repo, count))
-        except IndexError as e:  # FIXME: return to generic catch
-            exceptions[repo] = str(e)
-        except ImportError:
-            continue
-        finally:
-            if fp:
-                fp.close()
+            artifactory_policy = plugin_source.load_plugin(policy_name)
+        except ModuleNotFoundError:
+            if default:
+                LOG.info("No plugin found for %s. Applying Default", repo)
+                artifactory_policy = plugin_source.load_plugin('default')
+            else:
+                LOG.info("No plugin found for %s. Skipping Default", repo)
+                continue
+        artifacts = artifactory_policy.purgelist(
+            artifactory,
+            repo,
+            None,
+        )
+        count = artifactory.purge(repo, dryrun, artifacts)
+        LOG.info("processed {}, purged {}".format(repo, count))
 
     LOG.info("")
     LOG.info("Purging Performance:")
-    after = artifactory.list(reponame)
+    after = artifactory.list(None)
     for repo, info in after.items():
```

Comment:
Not specific to actual changes, but moving this to a separate function may make sense.

Reply:
completely agree, this was Alex code that I just had not refactored yet.